### DISCUSSION TITLE
AdGuard Home v0.107.52 on a DietPi reinstall the service?

### DISCUSSION THREAD

**Discussion by melucas** [AdguardTeam/AdGuardHome#7113] [Q&A] [answered]
Hello All,

I originally installed AdGuard Home using the `dietpi-software` manager a few years ago.

I've recently upgraded my AdGuard Home to version v0.107.52 using the AdGuard web interface.

The [release notes](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.52) mention reinstalling the service.  However, when I try `/mnt/dietpi_userdata/adguardhome/AdGuardHome -s uninstall`, I get the following error:

_root@AdGuard1:/mnt/dietpi_userdata/adguardhome# /mnt/dietpi_userdata/adguardhome/AdGuardHome -s uninstall
2024/07/05 13:22:16 [info] AdGuard Home, version v0.107.52
2024/07/05 13:22:16 [info] service: control action: uninstall
2024/07/05 13:22:16 [fatal] service: executing action "uninstall": Failed to uninstall AdGuard Home service: exit status 1_

When I run `journalctl -u adguardhome.service` the service is running and I can access it via the web ui.
 
Do I need to reinstall the service?  If so, what is the correct way to do that?

Thank you,
Michael

**Comment by EugeneOne1** [comment]
@melucas, hello.  It's required to stop the service before uninstalling it on some operating systems.  You can do that with the following command:

```sh
/mnt/dietpi_userdata/adguardhome/AdGuardHome -s stop
```

If that worked?

**Reply by melucas** [reply]
Thank you for the quick response.

I get the following error when trying to stop the service:

_2024/07/05 08:43:17 [info] AdGuard Home, version v0.107.52
2024/07/05 08:43:17 [info] service: control action: stop
2024/07/05 08:43:17 [fatal] service: executing action "stop": Failed to stop AdGuard Home service: exit status 5_

**Reply by EugeneOne1** [reply]
@melucas, oh, it seems I've missed the fact that AdGuard Home actually stops itself on "uninstall" command.  While it's generally safe to perform these command separately, the issue actually isn't related, apologies for the confusion.

Also, I didn't notice you've mentioned that the `dietpi-software` scripts were used.  Looking through the scripts I've found they are using their own custom service file avoiding the ones provided by AdGuard Home.  Thus, in your case no additional actions needed, it should work after update as intended.

**Reply by melucas** [reply, marked as answer]
@EugeneOne1, thank you for looking into this for me.

Everything is working as expected.  I will make this thread as answered.

Michael
